overlord/assertstate/assertstate.go

```diff
@@ -478,3 +478,16 @@ func ValidationSetAssertionForMonitor(st *state.State, accountID, name string, s
 	}
 	return as, false, err
 }
+
+// TemporaryDBWithModel returns a temporary database stacked on top of the
+// assertions database, with the provided model added to it. This is useful
+// during remodel, when the new model is not added to the main database until
+// the very last stage of the process.
+func TemporaryDBWithModel(st *state.State, model *asserts.Model) (asserts.RODatabase, error) {
+	db := cachedDB(st)
+	temp := db.WithStackedBackstore(asserts.NewMemoryBackstore())
+	if err := temp.Add(model); err != nil {
```

Comment:
needing to add the model here is a bit weird, because the writer does save the model, but the fetcher we define in devicestate/systems.go is read-only.

Reply:
I think my  point is that this probably should just be a TemporaryDB returning a full Database, and the Add should happen in devicestate with comments

Reply:
Updated